quarkus/runtime/src/main/java/org/keycloak/quarkus/runtime/configuration/mappers/CachingPropertyMappers.java

```diff
@@ -55,6 +55,7 @@ public static PropertyMapper<?>[] getClusteringPropertyMappers() {
                         .paramLabel("file")
                         .build(),
                 fromOption(CachingOptions.CACHE_EMBEDDED_MTLS_ENABLED)
+                        .isEnabled(CachingPropertyMappers::getDefaultMtlsEnabled, "a TCP based cache-stack is used")
```

Comment:
don't you need this option available with UDP to disable it?

Reply:
Also, it can be disabled when the method `org.keycloak.infinispan.util.InfinispanUtils#isRemoteInfinispan()` returns true because there is no transport.

Reply:
> don't you need this option available with UDP to disable it?

I found that it is 'false' when it is disabled, which is what we need. It seems we are lucky here? At least it worked nicely when I tested `jdbc-ping-udp` and then I didn't need to specify any additional option to disable it.

Reply:
I'm surprised the default value set in `CachingOptions` is not respected. 

Another point is that the stack name is not a good indicator of what kind of transport is used; it can be anything. The user can create a stack named `encrypted` with `UDP` + `ASYM_ENCRYPT` protocols, and this check will fail. 

It shouldn't cause any harm except creating certificates that won't be used :smile:

Reply:
Checking the cache name is for those stacks we know that they don't support TLS, and to avoid an exception later. By doing this, we make this backwards compatible for those you use a one of the well-known UDP configurations in KC 26.1 and earlier. 

In the docs I mention that people would need to disable the encryption - this is especially true for all custom UDP stacks. This is not ideal, but AFAIK sufficient to claim backwards compatibility.